enhancements/network/ovn-kubernetes-evpn.md

```diff
@@ -0,0 +1,292 @@
+
+---
+
+# OVN-Kubernetes support for EVPN
+
+## Summary
+
+This enhancement describes how OVN-Kubernetes support for EVPN integrates in
+OpenShift Container Platform (OCP). The feature allows exposing primary Cluster
+User Defined Networks (P-CUDNs) externally via a VPN to other entities either
+inside, or outside the cluster; using BGP and EVPN as a common and native
+networking standard that will enable integration into user networks without SDN
+specific network protocol integration, and providing an industry standardized
+way to achieve network segmentation between sites.
+
+OVN-Kubernetes support for EVPN is described in detail in an upstream
+[enhancement][1]. This document outlines the necessary changes to integrate it
+in OCP including the interaction with the Cluster Network Operator (CNO)
+and our test plan for this feature.
+
+[1]: https://github.com/ovn-kubernetes/ovn-kubernetes/pull/5089
+
+
+## Motivation
+
+The motivations for this feature are to be found in the upstream
+[enhancement][2].
+
+[2]: https://github.com/ovn-kubernetes/ovn-kubernetes/blob/8461a5526377488bc643bd4eb7024b0b735e8830/docs/okeps/okep-50088-evpn.md#introduction
+
+
+### User Stories
+
+The user stories for this feature are to be found in the upstream
+[enhancement][3].
+
+[3]: https://github.com/ovn-kubernetes/ovn-kubernetes/blob/8461a5526377488bc643bd4eb7024b0b735e8830/docs/okeps/okep-50088-evpn.md#user-storiesuse-cases
+
+### Goals
+
+The goals for this feature are to be found in the upstream [enhancement][4].
+
+[4]: https://github.com/ovn-kubernetes/ovn-kubernetes/blob/8461a5526377488bc643bd4eb7024b0b735e8830/docs/okeps/okep-50088-evpn.md#goals
+
+### Non-Goals
+
+The non-goals for this feature are to be found in the upstream [enhancement][5].
+
+[5]: https://github.com/ovn-kubernetes/ovn-kubernetes/blob/8461a5526377488bc643bd4eb7024b0b735e8830/docs/okeps/okep-50088-evpn.md#non-goals
+
+## Proposal
+
+The overall proposal and design for this feature is to be found in the upstream [enhancement][6].
+
+[6]: https://github.com/ovn-kubernetes/ovn-kubernetes/blob/8461a5526377488bc643bd4eb7024b0b735e8830/docs/okeps/okep-50088-evpn.md#proposed-solution
+
+The EVPN feature is mainly driven by the already existing OVN-Kubernetes
+RouteAdvertisements CRD. The `routeAdvertisements` configuration flag in the
+OVN-Kubernetes CNO configuration will need to be set to `Enabled` to be able to
+use the feature.
+
+Additionally, the EVPN feature is used through OVN-Kubernetes specific APIs in
+the form of new and updated CRDs. A new feature gate will be introduced for the
+EVPN feature and CNO will deploy these CRD updates only if the feature gate is
+enabled making it impossible to use the feature if the feature gate is not
+enabled.
+
+The EVPN feature is only supported when `routingViaHost` is set to false, also
+known as local gateway mode. CNO will perform no validation, OVN-Kubernetes will
+be in charge of reporting invalid configurations in this regard.
```

Comment:
would you elaborate a bit on how will it report invalid configurations ? events (which object) / logs / ...

Reply:
I will add a link to the specific section on the upstream enhancement for this